arch/cpu/nrf/sys/soc-rtc.c

```diff
@@ -0,0 +1,212 @@
+void clock_update(void);
+
+/*---------------------------------------------------------------------------*/
+#define COMPARE_INCREMENT (RTIMER_SECOND / CLOCK_SECOND)
+#define MULTIPLE_256_MASK 0xFFFFFF00
+/*---------------------------------------------------------------------------*/
+#if CLOCK_SIZE != 4
+#error CLOCK_CONF_SIZE must be 4 (32 bit)
+#endif
+#if RTIMER_SECOND != 32768
+#error RIMTER_SECOND must be 32768 when using the NRF_RTC
+#endif
+
+#ifdef NRF_CLOCK_CONF_RTC_INSTANCE
+#define NRF_CLOCK_RTC_INSTANCE NRF_CLOCK_CONF_RTC_INSTANCE
+#else
+#define NRF_CLOCK_RTC_INSTANCE 0
+#endif
+
+/* Prototype of a function in clock.c. Called every time the handler fires */
+void clock_update(void);
+/*---------------------------------------------------------------------------*/
+/**< RTC instance used for platform clock */
+static const nrfx_rtc_t rtc = NRFX_RTC_INSTANCE(NRF_CLOCK_RTC_INSTANCE);
+/*---------------------------------------------------------------------------*/
+static rtimer_clock_t last_isr_time = 0;
+static clock_time_t rtc_max_clock_ticks = 0;
+static rtimer_clock_t rtc_max_rtimer_ticks = 0;
+static volatile uint32_t overflow = 0;
+/*---------------------------------------------------------------------------*/
+static void
+clock_handler(nrfx_clock_evt_type_t event)
+{
+  (void)event;
+}
+/*---------------------------------------------------------------------------*/
+/**
+ * @brief Function for handling the RTC<instance> interrupts
+ * @param int_type Type of interrupt to be handled
+ */
+static void
+rtc_handler(nrfx_rtc_int_type_t int_type)
+{
+  uint32_t next;
+  if(int_type == SOC_RTC_SYSTEM_CH) {
+    clock_update();
+    next = (nrfx_rtc_counter_get(&rtc) + COMPARE_INCREMENT) & MULTIPLE_256_MASK;
+    nrfx_rtc_cc_set(&rtc, SOC_RTC_SYSTEM_CH, next, true);
+  } else if(int_type == SOC_RTC_RTIMER_CH) {
+    /* rtimer event */
+    rtimer_run_next();
+    /* We need to handle the compare event */
+  } else if(int_type == NRFX_RTC_INT_OVERFLOW) {
+    overflow++;
+  }
+  last_isr_time = soc_rtc_get_rtimer_ticks();
+}
+/*---------------------------------------------------------------------------*/
+/**
+ * @brief Function starting the internal LFCLK XTAL oscillator.
+ */
+static void
+lfclk_config(void)
+{
+  nrfx_err_t err_code = nrfx_clock_init(clock_handler);
+
+  if(err_code != NRFX_SUCCESS) {
+    return;
+  }
+
+  nrfx_clock_enable();
+
+  nrfx_clock_lfclk_start();
+}
+/*---------------------------------------------------------------------------*/
+/**
+ * @brief Function initialization and configuration of RTC driver instance.
+ */
+static void
+rtc_config(void)
+{
+  nrfx_err_t err_code;
+
+  /*Initialize RTC instance */
+  nrfx_rtc_config_t config = NRFX_RTC_DEFAULT_CONFIG;
+  config.prescaler = RTC_FREQ_TO_PRESCALER(32768); /* full speed... */
+  config.interrupt_priority = 6;
+  config.reliable = 0;
+
+  err_code = nrfx_rtc_init(&rtc, &config, rtc_handler);
+
+  if(err_code != NRFX_SUCCESS) {
```

Comment:
This if-statment doesn't make any difference since no other code follows, so it can be removed.